core/src/main/java/io/methvin/watchservice/MacOSXListeningWatchService.java

```diff
@@ -145,6 +148,8 @@ public AbstractWatchKey register(
             flags);
 
     final CFRunLoopThread thread = new CFRunLoopThread(streamRef, file.toFile());
+    callback.onClose(() -> {watchKey.watchService().close(thread, callback, file); return null;});
```

Comment:
We don't use the return value, so you can change the type to `Runnable` to avoid the ugly `return null` here.

Reply:
done